src/components/ClipBoardButton.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from "react";
+import { FileCopy } from "@mui/icons-material";
+import Tooltip from "@mui/material/Tooltip";
+import IconButton from "@mui/material/IconButton";
+
+const ClipboardButton = ({ textToCopy }) => {
+  const [isCopied, setIsCopied] = useState(false);
```

Comment:
Currently you're setting `timeoutId` and never resetting it, only to track isCopied separately. Could we just infer `isCopied` from `!!timeoutId` and track only one state?